### ISSUE TITLE
NavigationMeshGenerator's bake method crashes the game

### ISSUE THREAD

**Issue by ic3bug** [godotengine/godot#36448]
**Godot version:**
Godot 3.2 stable

**OS/device including version:**
Windows 10

**Issue description:**
NavigationMeshGenerator's bake method crashes the game without throwing any error.

**Steps to reproduce:**

- Create a scene
- Create a navigation node
- Create a navigation mesh instance node as a child of navigation node
- Create a plane and other meshes representing a level
- Create a script that dynamically generates a navmesh
```
var nav = get_node("navigation")
var navmesh_instance = nav.get_node("navigation_mesh_instance")
var navmesh = navmesh_instance.navmesh
NavigationMeshGenerator.bake(navmesh, self)
nav.navmesh_add(navmesh, navmesh_instance.global_transform)
```
- Start a scene

**Minimal reproduction project:**
[Navmesh Test.zip](https://github.com/godotengine/godot/files/4239406/Navmesh.Test.zip)

**Comment by KawanWeege** [comment]
The bake function is not supported for in-game use in 3.2. It should only be used in for tool scripts.

The reason why it crashes is because it will try to display the editor's progress bar in gane and well, the game doesn't have this progress bar.

In-game baking is already supported in 4.0.

**Comment by ic3bug** [comment]
Ok, thanks. Closing the issue.

**Comment by akien-mga** [comment]
A crash is still bad though, so it would be good to abort cleanly instead of crashing in this case. It's fixed in `master` but a custom workaround for `3.2` would be welcome.

**Comment by akien-mga** [comment]
Fixed by #36470.
